controlplane/kubeadm/internal/controllers/remediation.go

```diff
@@ -146,6 +148,13 @@ func (r *KubeadmControlPlaneReconciler) reconcileUnhealthyMachines(ctx context.C
 			return ctrl.Result{}, nil
 		}
 
+		// The cluster MUST have no machines with a deletion timestamp. This rule prevents KCP taking actions while the cluster is in a transitional state.
```

Comment:
```suggestion
		// The cluster MUST have no machines being provisioned. This rule prevents KCP taking actions while the cluster is in a transitional state.
```

TBH we might not even need the comment. IMO the code below is self-explanatory, given the method name in the if clause and the log message.